templates/tfengine/components/project/main.tf

```diff
@@ -30,17 +30,22 @@ module "project" {
   billing_account = local.constants.billing_account
   {{- else}}
 
-  name                    = "{{.project_id}}"
+  name            = "{{.project_id}}"
   {{- if eq .parent_type "organization"}}
-  org_id                  = "{{.parent_id}}"
+  org_id          = "{{.parent_id}}"
   {{- else}}
-  org_id                  = ""
-  folder_id               = "{{.parent_id}}"
+  org_id          = ""
+  folder_id       = "{{.parent_id}}"
   {{- end}}
-  billing_account         = "{{.billing_account}}"
+  billing_account = "{{.billing_account}}"
   {{- end}}
-  lien                    = {{get . "enable_lien" true}}
+  lien            = {{get . "enable_lien" true}}
+  # Create and keep default service accounts when certain APIs are enabled.
   default_service_account = "keep"
+  {{- if not (get . "use_constants" false)}}
```

Comment:
Is this doing anything? Seems to only have a comment inside?

Reply:
Yea, i need this because if `use_contants` is `true`, there is no `project_id` input as the project ID is constructed from prefix and env code.

Reply:
Just want to dynamically construct this comment to include the project ID as well.

Reply:
Ah I see. I don't think it is that important to include the full SA name since it can be assumed it will be created in this project. What about changing the wording to be more generic e.g.

Do not create the additional project service account to run Terraform as it will be handled by separate CICD process.

Reply:
Done.

Rephrased a little as the SA is to be used for Compute Engine generically. https://github.com/terraform-google-modules/terraform-google-project-factory#features